### PR TITLE
feat(ST-171): implement student profile interface

### PR DESCRIPTION
**Pull request by akib-raihan-sazim**
## Description of Change
- Created StudentProfile component with interface file
- Updated ProfileContainer component
- 

## Associated Ticket Link(s)
- https://trello.com/c/IFguIYTt

## Screenshots / Screen Recordings

https://github.com/user-attachments/assets/4242d57e-486b-4569-b7e1-069d577273fa

### CODE REVIEW

File: modules/Profile/containers/ProfileContainer/ProfileContainer.tsx [hunk 1]
```diff
@@ -1,33 +1,51 @@
...
 import { useGetUserDetailsQuery } from "@/shared/redux/rtk-apis/users/users.api";
 
+import StudentProfile from "../../components/StudentProfile/StudentProfile";
 import TeacherProfile from "../../components/TeacherProfile/TeacherProfile";
 import { useStyles } from "./ProfileContainer.styles";
 
 const ProfileContainer = () => {
   const { classes } = useStyles();
-  const { data: user, isLoading, error } = useGetUserDetailsQuery();
+  const [isClient, setIsClient] = useState(false);
```
Comment (ishmam-iqbal-sazim) [inline modules/Profile/containers/ProfileContainer/ProfileContainer.tsx:18]: What is this for?

File: modules/Profile/components/StudentProfile/StudentProfile.tsx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+const StudentProfile: React.FC<IStudentProfileProps> = ({ user }) => {
+  if (!user || !user.student) {
+    notifications.show({
+      title: "Profile Data Unavailable",
+      message: "No teacher profile data is available at the moment.",
+      color: "red",
+    });
+  }
+  if (!user || !user.student) return null;
```
Comment (ishmam-iqbal-sazim) [inline modules/Profile/components/StudentProfile/StudentProfile.tsx:16]: Same comments as previous PR

File: modules/Profile/components/StudentProfile/StudentProfile.tsx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  const profileItems = [
+    { label: "Email", value: user.email },
+    { label: "Gender", value: user.gender || "Not specified" },
+    { label: "First name", value: user.firstName },
+    { label: "Last name", value: user.lastName },
+    { label: "Education Level", value: user.student.educationLevel },
+    { label: "Phone Number", value: user.student.phoneNo },
+    { label: "Address", value: user.student.address },
+  ];
```
Comment (ishmam-iqbal-sazim) [inline modules/Profile/components/StudentProfile/StudentProfile.tsx:26]: Should be inside a helpers.ts file

File: modules/Profile/components/StudentProfile/StudentProfile.tsx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      { label: "Medium", value: user.student.medium || "Not specified" },
+      { label: "Class", value: user.student.class || "Not specified" },
+    );
+  } else if (user.student.educationLevel === EEducationLevel.UNIVERSITY) {
+    profileItems.push(
+      { label: "Degree Type", value: user.student.degree || "Not specified" },
+      { label: "Degree Name", value: user.student.degreeName || "Not specified" },
+      { label: "Semester/Year", value: user.student.semester || "Not specified" },
+    );
+  }
```
Comment (ishmam-iqbal-sazim) [inline modules/Profile/components/StudentProfile/StudentProfile.tsx:42]: You can also incorporate this logic inside the helper

File: modules/Profile/containers/ProfileContainer/ProfileContainer.tsx (hunk 1 above)
Comment (akib-raihan-sazim) [reply modules/Profile/containers/ProfileContainer/ProfileContainer.tsx]: I have done it in another way. I am fetching the `currentUser` and then checking if the currentUser.userId exist.

**Review by ragib-hossain-sazim — APPROVED** [review]

File: modules/Profile/components/StudentProfile/StudentProfile.tsx [hunk 5]
```diff
@@ -0,0 +1,31 @@
+import { Grid, Title } from "@mantine/core";
+
+import { getStudentProfileItems } from "./StudentProfile.helpers";
+import { IStudentProfileProps } from "./StudentProfile.interface";
+
+const StudentProfile: React.FC<IStudentProfileProps> = ({ user }) => {
+  if (!user || !user.student) return null;
```
Comment (ishmam-iqbal-sazim) [inline modules/Profile/components/StudentProfile/StudentProfile.tsx:7]: Can remove this for the same reasons mentioned in previous PR

**Review by ishmam-iqbal-sazim — APPROVED** [review]
Approved pending comments

**Pull request merged**